torch/_inductor/select_algorithm.py

```diff
@@ -849,47 +846,48 @@ def benchmark_in_current_process(choice):
             torch.cuda.synchronize()  # shake out any CUDA errors
             return result
 
-        def benchmark_in_sub_process(choice):
-            # only benchmark triton kernel in sub process for now.
-            # ATen/Extern kernel are still benchmarked in the current process.
-            if isinstance(choice, ExternKernelCaller):
-                return benchmark_in_current_process(choice)
-
+        def benchmark_in_current_process(options):
+            timings = {}
+            for choice in options:
+                try:
+                    timing = benchmark_choice_in_current_process(choice)
+                except RuntimeError as e:
+                    msg = str(e)
+                    if "invalid argument" in msg:
+                        msg += "\n\nThis may mean this GPU is too small for max_autotune mode.\n\n"
+                        log.warning(msg)
+                        timing = float("inf")
+                    else:
+                        if "illegal memory access" in msg:
+                            msg += "\n\nEither error in template or triton bug.\n"
+                        raise ErrorFromChoice(msg, choice, debug_str())
+                except AssertionError as e:
+                    raise AssertionError(
+                        f"Incorrect result from choice {choice}\n\n{e}"
+                    )
+
+                timings[choice] = timing
+
+            return timings
+
+        def benchmark_in_sub_process(options):
             from . import autotune_process
 
-            if DEBUG:
```

Comment:
The benchmarking all happens in parallel now, so I didn't think the timestamp info was as relevant.

Reply:
Hmm, it would still be helpful to know how long each benchmark request takes to be processed?

Reply:
@shunting314 do you mean the overall parallel benchmark request in aggregate? Or do you mean each individual request? I wondered if the existing logging was already providing that. For example, I'm seeing this whenever I enable max autotune:

AUTOTUNE mm(512x768, 768x50265)
  triton_mm_1586 0.2621 ms 100.0%
  triton_mm_1585 0.2653 ms 98.8%
  triton_mm_1591 0.3040 ms 86.2%
  triton_mm_1587 0.3044 ms 86.1%
  triton_mm_1588 0.3108 ms 84.3%
  triton_mm_1584 0.3300 ms 79.4%
  triton_mm_1592 0.4615 ms 56.8%
  triton_mm_1594 0.4911 ms 53.4%
  triton_mm_1593 0.5753 ms 45.6%
  triton_mm_1590 0.6271 ms 41.8%
SubProcess AUTOTUNE takes 5.7510 seconds